### PR TITLE
chore: caching refactor

### PR DESCRIPTION
**Pull request by decleaver**
## Description

Standardizing the way caching is being implemented

## Related Issue

Relates to #653 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Other (security config, docs update, etc)

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [ ] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-cli/blob/main/CONTRIBUTING.md) followed

### CODE REVIEW

File: src/pkg/bundle/pull.go [hunk 1]
```diff
@@ -25,11 +26,9 @@ import (
...
 func (b *Bundle) Pull() error {
 	ctx := context.TODO()
-	// use uds-cache/packages as the dst dir for the pull to get auto caching
-	// we use an ORAS ocistore to make that dir look like an OCI artifact
-	cacheDir := filepath.Join(zarfConfig.GetAbsCachePath(), "packages")
-	if err := helpers.CreateDirectory(cacheDir, 0o755); err != nil {
-		return err
+	tmpDstDir, err := zarfUtils.MakeTempDir(config.CommonOptions.TempDirectory)
+	if err != nil {
+		return fmt.Errorf("pull bundle unable to create temp directory: %w", err)
```
Comment (UncleGedd) [inline src/pkg/bundle/pull.go:31]:
```suggestion
		return fmt.Errorf("unable to create temp directory: %w", err)
```

❤️ the use of `%w`, TIL

File: src/pkg/bundle/remote.go [hunk 2]
```diff
@@ -15,18 +15,19 @@ import (
...
 	"github.com/defenseunicorns/zarf/src/pkg/message"
-	zarfUtils "github.com/defenseunicorns/zarf/src/pkg/utils"
 	"github.com/defenseunicorns/zarf/src/pkg/zoci"
 	"github.com/mholt/archiver/v4"
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 	"golang.org/x/exp/slices"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"oras.land/oras-go/v2"
+
+	// "oras.land/oras-go/v2"
```
Comment (UncleGedd) [inline src/pkg/bundle/remote.go:30]: stray comment

File: src/pkg/cache/layers.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	"github.com/defenseunicorns/uds-cli/src/config"
+	"github.com/defenseunicorns/uds-cli/src/pkg/utils/boci"
+	zarfUtils "github.com/defenseunicorns/zarf/src/pkg/utils"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2"
+	ocistore "oras.land/oras-go/v2/content/oci"
+	"oras.land/oras-go/v2/registry/remote"
+)
+
+// checkLayerExists checks if a layer already exists in the bundle store or the cache, copying it to the dstDir if it does
```
Comment (UncleGedd) [inline src/pkg/cache/layers.go:23]:
```suggestion
// CheckLayerExists checks if a layer already exists in the bundle store or the cache, copying it to the dstDir if it does
```

File: src/pkg/bundle/pull.go (hunk 1 above)
Comment (decleaver) [reply src/pkg/bundle/pull.go:31]: yeah, not really utilizing it right now but could help out in the future

**Review by UncleGedd — APPROVED** [review]
paired and refactored together

**Review by TristanHoladay — APPROVED** [review]

**Pull request merged**